File: ex4.js
function Person(tasks) {
    if (!(tasks instanceof Array)) {
        throw 'tasks must be an Array'
    }

    this.tasks = tasks;
};

Person.prototype.completedTasks = function () {
    return this.tasks.filter(function (task) {
        return task.completed
    }).join("\n")
};

function Tasks(name, completed) {
    this.name = name;
    this.completed = completed;
};

Tasks.prototype.toString = function () {
    return this.name + ' (' + this.completed + ')'
};

var task1 = new Tasks('ex. 1', true);
var task2 = new Tasks('ex. 2', false);
var task3 = new Tasks('ex. 3', true);
var task4 = new Tasks('ex. 4', false);
var task5 = new Tasks('ex. 5', false);
var task6 = new Tasks('ex. 6', true);

console.log(new Person([task1, task2, task3, task4, task5, task6]).completedTasks());

File: ex4.rb
class Person
  attr_accessor :tasks

  def initialize(tasks)
    raise 'tasks must be an array' unless tasks.is_a?(Array)
    @tasks = tasks
  end

  def completed_tasks
    tasks.select { |task| task.completed }
  end
end

class Tasks
  attr_accessor :name, :completed

  def initialize(name, completed)
    @name = name
    @completed = completed
  end

  def to_s
    "#{name} (#{completed})"
  end
end


task1 = Tasks.new('ex. 1', true)
task2 = Tasks.new('ex. 2', false)
task3 = Tasks.new('ex. 3', true)
task4 = Tasks.new('ex. 4', false)
task5 = Tasks.new('ex. 5', false)
task6 = Tasks.new('ex. 6', true)

puts Person.new([task1, task2, task3, task4, task5, task6]).completed_tasks
